splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/sequence/AbstractSequence.java

```diff
@@ -52,6 +52,10 @@ public long getNext() throws StandardException{
         return currPosition.getAndAdd(incrementSteps);
     }
 
+    public long peekAtCurrentValue() throws StandardException {
+        return currPosition.get();
```

Comment:
The current block can only be consumed by getNext(). PeekAtCurrentValue does not consume anything, only returns the current value.